syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";

import "protos/base.proto";

enum InterpreterType {
  UNKNOWN_I = 0;
  BASIC_I = 1;
  STATISTICAL_I = 2;
  LLM_CHAT_GPT_VISION_I = 3;
}


message Interpretation {
  enum Type {
    UNKNOWN = 0;
    IMAGE = 1;
    TEXT = 2;
    CSV_FILE = 3;
    JSON = 4;
  }

  enum ModelType {
    UNKNOWN_M = 0;
    PLAYBOOK_TASK = 1;
    PLAYBOOK_STEP = 2;
    PLAYBOOK_STEP_RELATION = 3;
    WORKFLOW_EXECUTION = 4;
  }

  Type type = 1;
  InterpreterType interpreter_type = 2;
  google.protobuf.StringValue title = 3;
  google.protobuf.StringValue description = 4;
  google.protobuf.StringValue summary = 5;
  google.protobuf.StringValue image_url = 6;
  google.protobuf.StringValue file_path = 7;
  google.protobuf.StringValue object_url = 8;
  ModelType model_type = 9;
  google.protobuf.StringValue object_uid = 10;
}